pkg/controllers/core/deployer_controller.go

```diff
@@ -613,8 +636,37 @@ func (r *DeployerReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Owns(&corev1.Service{}).
 		Owns(&networkingv1beta1.Ingress{}).
 		// watch for build mutations to update dependent deployers
+		Watches(&source.Kind{Type: &corev1.ConfigMap{}}, r.enqueueTrackedConfigMapResource()).
 		Watches(&source.Kind{Type: &buildv1alpha1.Application{}}, enqueueTrackedResources(&buildv1alpha1.Application{})).
 		Watches(&source.Kind{Type: &buildv1alpha1.Container{}}, enqueueTrackedResources(&buildv1alpha1.Container{})).
 		Watches(&source.Kind{Type: &buildv1alpha1.Function{}}, enqueueTrackedResources(&buildv1alpha1.Function{})).
 		Complete(r)
 }
+
+func (r *DeployerReconciler) constructHostForDeployer(deployer *corev1alpha1.Deployer, cm corev1.ConfigMap) string {
```

Comment:
move to be closer to where this method is used. Just below `constructLabelsForDeployer` would be a good spot.